app/src/main/java/com/imashnake/animite/features/searchbar/SearchFrontDrop.kt

```diff
@@ -35,42 +37,69 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.res.vectorResource
 import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.lifecycle.viewmodel.compose.viewModel
 import androidx.navigation.NavHostController
 import com.imashnake.animite.SearchQuery
+import com.imashnake.animite.features.appCurrentDestinationAsState
 import com.imashnake.animite.features.destinations.MediaPageDestination
+import com.imashnake.animite.features.navigationbar.NavigationBarPaths
 import com.imashnake.animite.type.MediaType
 import com.ramcosta.composedestinations.navigation.navigate
 import com.imashnake.animite.R as Res
 
 // TODO:
 //  - UX concern: This blocks content sometimes!
-//  - `SearchList` goes beyond the status bar.
-@OptIn(ExperimentalAnimationApi::class, ExperimentalMaterial3Api::class)
+//  - Create a custom layout for `FrontDrop`s.
+@OptIn(
+    ExperimentalAnimationApi::class,
+    ExperimentalMaterial3Api::class,
+    ExperimentalLayoutApi::class
+)
 @Composable
-fun SearchBar(
+fun SearchFrontDrop(
     modifier: Modifier,
     viewModel: SearchViewModel = viewModel(),
     navController: NavHostController
 ) {
     var isExpanded by remember { mutableStateOf(false) }
+    var isMaskVisible by remember { mutableStateOf(false) }
+    val maskAlpha: Float by animateFloatAsState(
+        targetValue = if (isExpanded) 0.95f else 0f,
+        animationSpec = tween(500),
+        // Mask waits for the alpha to decrease.
+        finishedListener = {
+            isMaskVisible = false
+        }
+    )
+
+    // TODO: This hack still results in a very slight jitter (still an improvement); remove this
+    //  after we start using custom `Scaffold`s.
+    val searchBarBottomPadding: Dp by animateDpAsState(
+        targetValue = dimensionResource(Res.dimen.large_padding) + if (
+            NavigationBarPaths.values().any {
+                it.direction == navController.appCurrentDestinationAsState().value
+            } && !isExpanded
+        ) dimensionResource(Res.dimen.navigation_bar_height) else 0.dp,
+        animationSpec = tween(delayMillis = 100)
+    )
 
-    Column(modifier = modifier) {
-        // TODO: Customize this animation.
-        AnimatedContent(targetState = isExpanded) { targetExpanded ->
-            if (targetExpanded) {
+    if(isExpanded || isMaskVisible) {
+        Box(
+            Modifier
+                .background(MaterialTheme.colorScheme.background.copy(alpha = maskAlpha))
+                .fillMaxSize()
+        ) {
+            if(isExpanded) {
                 SearchList(
                     viewModel = hiltViewModel(),
```

Comment:
why we passing in a ViewModel?

Reply:
Agreed. Makes more sense to pass a list to SearchList instead